ex01/app.js

```diff
@@ -0,0 +1,81 @@
+// ORIGINAL SPEC:
+// Write a function that takes a positive integer as an argument &
+// returns a Boolean indicating whether the argument is a power of three.
+// Hint: you might want to use recursion for this, which we haven't talked about yet. Exciting!
+// 3, 9, 27, 81...
+
+"use strict";
+
+// Define UI Vars
+const form = document.querySelector("#number-form");
+const numberInput = document.querySelector("#number-input");
+const bigGuess = document.querySelector("#guess");
+const guessList = document.querySelector("#guess-list");
+let recents = [];
+
+// event listener is on the 'submit' part of the form.
+form.addEventListener("submit", checkNum);
+
+// f() checkNum - throws an alert() for an empty form or calls showNumber():
+function checkNum(e) {
+  e.preventDefault(); // not sure this is the best approach?  It's the only way i could find to stop the Form from refreshing the page and killing my HTML edits.
+  if (numberInput.value === "") {
+    alert("Type any Integer to test if it's a power of 3");
+  } else {
+    if (threePower(1, numberInput.value)) {
+      //console.log("true");
+      showNumber(true, numberInput.value);
+    } else {
+      //console.log("false");
+      showNumber(false, numberInput.value);
+    }
+  }
+}
+
+// f() threePower - This function uses recurssion to check if the user input is a power of 3:
```

Comment:
The recursive approach will work. There's also iterative solutions, e.g. just compute powers of 3 until (computed number) is greater than or equal to the input value. A more advanced solution will precompute a bunch of powers of 3 and store those, then add to the list as it gets bigger and bigger inputs